src/pages/settings/Security/TwoFactorAuth/TwoFactorAuthPage.js

```diff
@@ -0,0 +1,81 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import {withOnyx} from "react-native-onyx";
+import PropTypes from "prop-types";
+import CodesStep from "./Steps/CodesStep";
+import DisableStep from "./Steps/DisableStep";
+import IsEnabledStep from "./Steps/IsEnabledStep";
+import VerifyStep from "./Steps/VerifyStep";
+import SuccessStep from "./Steps/SuccessStep";
+import ONYXKEYS from "../../../../ONYXKEYS";
+import CONST from "../../../../CONST";
+import * as TwoFactorAuthActions from '../../../../libs/actions/TwoFactorAuthActions';
+import TwoFactorAuthContext from "./TwoFactorAuthContext";
+
+const propTypes = {
+    account: PropTypes.shape({
+        /** Whether or not the user has two factor authentication enabled */
+        requiresTwoFactorAuth: PropTypes.bool,
+
+        /** The current step in the two factor authentication process */
+        twoFactorAuthStep: PropTypes.string,
+    }),
+}
+
+const defaultProps = {
+    account: {
+        requiresTwoFactorAuth: false,
+        twoFactorAuthStep: '',
+    }
+}
+
+function TwoFactorAuthPage({account,}) {
+    const [currentStep, setCurrentStep] = useState(CONST.TWO_FACTOR_AUTH_STEPS.CODES);
+
+    useEffect(() => {
+        if (account.twoFactorAuthStep) {
+            setCurrentStep(account.twoFactorAuthStep);
+            return
+        }
+        if (account.requiresTwoFactorAuth) {
+            setCurrentStep(CONST.TWO_FACTOR_AUTH_STEPS.IS_ENABLED);
+        } else {
+            setCurrentStep(CONST.TWO_FACTOR_AUTH_STEPS.CODES);
+        }
+    }, [account.requiresTwoFactorAuth]);
```

Comment:
Deps. array should include all dependencies.
```suggestion
    }, [account.requiresTwoFactorAuth, account.twoFactorAuthStep]);
```

Reply:
I don't want to fire it everytime the step changes

Reply:
No, there won't be any collision and there won't be different value. When changing state, you're changing Onyx. There won't be a scenario that you trigger the change only in the Onyx

Reply:
Okay 👍